yarn2-plugins/prune-dev-dependencies/src/index.ts

```diff
@@ -1,9 +1,20 @@
-import { Cache, Configuration, Plugin, Project, StreamReport, CommandContext } from '@yarnpkg/core'
-import { Command } from 'clipanion'
+import {
+  Cache,
+  Configuration,
+  Plugin,
+  Project,
+  StreamReport,
+  CommandContext,
+  YarnVersion
+} from '@yarnpkg/core'
+import { decorateClass } from './utils'
 
-class HerokuPruneDevDependenciesCommand extends Command<CommandContext> {
-  @Command.Path('heroku', 'prune')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const clipanion = require('clipanion')
+const version = YarnVersion ?? '0.0.0'
+const YARN_4 = /^4\./.test(version)
```

Comment:
In my other PR, I future proofed it for future versions beyond v4, might be worth including.

Reply:
yes, i did notice that you future-proofed up to v9 @RDIL 😉 

but i'm a bit hesitant to assume that v5 would "just work" since every time Yarn releases a new version some type of change is required.  i'd rather be explicit.  but let me think about that some more since i'll need to add a test to cover that scenario anyways.

Reply:
@RDIL i modified the version check + compatibility changes to instead:
* attempt to configure the command with the clipanion annotation (works for Yarn 2 & 3)
* if that fails, attempt to configure the command paths (works in Yarn 4 and maybe future versions)
* log an error if none of the strategies work